packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -437,6 +472,28 @@ File _findApkFile(GradleProject project, BuildInfo buildInfo) {
   return null;
 }
 
+File _findBundleFile(GradleProject project, BuildInfo buildInfo) {
+  final String bundleFileName = project.bundleFileFor(buildInfo);
+
+  if (bundleFileName == null)
+    return null;
+  File bundleFile = fs.file(fs.path.join(project.bundleDirectory.path, bundleFileName));
+  if (bundleFile.existsSync()) {
+    return bundleFile;
+  }
+  final String modeName = camelCase(buildInfo.modeName);
+  bundleFile = fs.file(fs.path.join(project.bundleDirectory.path, modeName, bundleFileName));
+  if (bundleFile.existsSync())
+    return bundleFile;
+  if (buildInfo.flavor != null) {
+    // Android Studio Gradle plugin v3 adds flavor to path. for bundle the folder name is flavorModename
```

Comment:
nit: would write this as `Android Studio Gradle plugin v3 adds the flavor to the path.  For the bundle the folder name is the flavor plus the mode name.`